api/v1alpha1/types_etcd_test.go

```diff
@@ -125,19 +97,164 @@ var _ = Describe("Etcd", func() {
 				Controller:         pointer.Bool(true),
 				BlockOwnerDeletion: pointer.Bool(true),
 			}
-			Expect(created.GetAsOwnerReference()).To(Equal(expected))
+			Expect(etcd.GetAsOwnerReference()).To(Equal(expected))
 		})
 	})
 
 	Context("GetRoleName", func() {
 		It("should return the role name for the Etcd", func() {
-			Expect(created.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+			Expect(etcd.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
 		})
 	})
 
 	Context("GetRoleBindingName", func() {
 		It("should return the rolebinding name for the Etcd", func() {
-			Expect(created.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+			Expect(etcd.GetRoleName()).To(Equal(GroupVersion.Group + ":etcd:foo"))
+		})
+	})
+
+	Context("IsBackupStoreEnabled", func() {
+		Context("when backup is enabled", func() {
+			It("should return true", func() {
+				Expect(etcd.IsBackupStoreEnabled()).To(Equal(true))
+			})
+		})
+		Context("when backup is not enabled", func() {
+			It("should return false", func() {
+				etcd.Spec.Backup = BackupSpec{}
+				Expect(etcd.IsBackupStoreEnabled()).To(Equal(false))
+			})
+		})
+	})
+
+	Context("IsMarkedForDeletion", func() {
+		Context("when deletion timestamp is not set", func() {
+			It("should return false", func() {
+				Expect(etcd.IsMarkedForDeletion()).To(Equal(false))
+			})
+		})
+		Context("when deletion timestamp is set", func() {
+			It("should return true", func() {
+				etcd.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+				Expect(etcd.IsMarkedForDeletion()).To(Equal(true))
+			})
+		})
+	})
+
+	Context("GetSuspendEtcdSpecReconcileAnnotationKey", func() {
+		ignoreReconciliationAnnotation := IgnoreReconciliationAnnotation
+		suspendEtcdSpecReconcileAnnotation := SuspendEtcdSpecReconcileAnnotation
+		Context("when etcd has only annotation druid.gardener.cloud/ignore-reconciliation set", func() {
+			It("should return druid.gardener.cloud/ignore-reconciliation", func() {
+				etcd.Annotations = map[string]string{
+					IgnoreReconciliationAnnotation: "true",
+				}
+				Expect(etcd.GetSuspendEtcdSpecReconcileAnnotationKey()).To(Equal(&ignoreReconciliationAnnotation))
+			})
+		})
+		Context("when etcd has only annotation druid.gardener.cloud/suspend-etcd-spec-reconcile set", func() {
+			It("should return druid.gardener.cloud/suspend-etcd-spec-reconcile", func() {
+				etcd.Annotations = map[string]string{
```

Comment:
this is what I'm refering to in above comment

Reply:
The strings here are printed when tests are executed, and make it easy to read the tests. If you prefer, we can use an `fmt.Sprintf()` here and use the constants for generating the test description string instead, which seems unnecessary to me though. WDYS?

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/c0895d233daee0e4aba48a2740cba1b90161adfd#diff-ff02116a8c95e5d174caa335f023a4b68536586768589966283710a85d8e4cbd